### PR TITLE
DOCS-9375 adding content for custom ingress config.

### PR DESCRIPTION
**Pull request by sleslie23**

### CODE REVIEW

File: modules/ROOT/pages/custom-ingress-configuration.adoc [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+** `rules.http.paths.path`: Specifies the `path` value used in the `http rules` section of the Ingress resource. 
++
+Generally, this value corresponds to the public URL of a Mule application shown in the application URL preview. To allow this value to be configured dynamically during deployment, use `/$(APP_PATH)` as a placeholder.
+
+. Run `rtfctl apply ingress-configmap <custom-template-file>`, using the file you just created as the value for `custom-template-file`.
+
+After running this command `rtfctl` validates the template file you provided, then performs the following:
+
+* Creates the ConfigMap file in the `rtf` namespace. Any errors are written to stdout.
+* Schedules a job to be run in the cluster. This job calls the `rtf-agent` endpoint which validates the `ingressResouce` using data you specified in the template file.
```
Comment (sundippatel) [inline modules/ROOT/pages/custom-ingress-configuration.adoc]: I think we could shorten this to "Schedules a job to validate the ingress resource generated from the config map in the template file."

File: modules/ROOT/pages/custom-ingress-configuration.adoc [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+* Waits until the job completes, then views the pod logs for the job status. `rtfctl` displays the status to stdout.
```
Comment (sundippatel) [inline modules/ROOT/pages/custom-ingress-configuration.adoc]: "..job completes to output the status and relevant logs."

**Review by sundippatel — COMMENTED** [review]
Like this; seems easy to follow.
I think we could add a note letting the user know the following:
 a) if they make changes to the template, they will need to ask their MuleSoft organization admin to redeploy apps to pick up the new ingress changes.
 b) the "rtfctl apply" command will not make changes to the template used by the RTF agent if the validation returns an error.

File: modules/ROOT/pages/custom-ingress-configuration.adoc [hunk 3]
```diff
@@ -0,0 +1,52 @@
+= Defining a Custom Ingress Configuration
+
+[NOTE]
+====
+The tasks outline in this topic must be performed by an IT administrator
+====
+
+Runtime Fabric on Self-managed Kubernetes enables you to edit a template file that specifies a custom Ingress configuration. Using this file you can define annotations, rewrite rules, the ingressClassName, and more.
+
+The template file is a Kubernetes ConfigMap file. After customizing the template, you can use the `rtfctl apply` command-line utility to apply the custom Ingress configuration to your Kubernetes cluster.
```
Comment (kmendiratta-crm) [inline modules/ROOT/pages/custom-ingress-configuration.adoc]: we might want to give the full sub-command `rtfctl apply ingress-configmap`

File: modules/ROOT/pages/custom-ingress-configuration.adoc [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    kubernetes.io/ingress.class: <name>
+  ingress.spec: |
+    ingressClassName: <name>
+    rules.http.paths.path: /${APP_PATH}
+----
+
+. Edit the text file as required.
++
+All changes should be made under the `data` section of this file. Do not modify any other sections of the template. The most common configuration options are.
++
```
Comment (kmendiratta-crm) [inline modules/ROOT/pages/custom-ingress-configuration.adoc:36]: Can we tabulate this in a manner consistent with what we do elsewhere, for e.g.: https://docs.mulesoft.com/runtime-fabric/1.7/runtime-fabric-logs#azure-log-analytics-configuration-parameters

cc @chingyi-lin

File: modules/ROOT/pages/custom-ingress-configuration.adoc [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+[NOTE]
+====
+The tasks outline in this topic must be performed by an IT administrator
+====
+
+Runtime Fabric on Self-managed Kubernetes enables you to edit a template file that specifies a custom Ingress configuration. Using this file you can define annotations, rewrite rules, the ingressClassName, and more.
+
+The template file is a Kubernetes ConfigMap file. After customizing the template, you can use the `rtfctl apply` command-line utility to apply the custom Ingress configuration to your Kubernetes cluster.
+
+If you do not apply a custom template file, Runtime Fabric uses a default configuration that works with the out-of-the-box Kubernetes Ingress configuration.
```
Comment (kmendiratta-crm) [inline modules/ROOT/pages/custom-ingress-configuration.adoc]: This may seem slightly confusing. What we really we want to communicate to our customers is:
1. We (RTF) tries to create a kubernetes default ingress resource if no configmap data is applied/configured.
2. The default ingress resource may/may not work depending on the app-path and the ingress controller being used by the customer. For e.g. if the app endpoint expects subpaths in the request, and the customer is using nginx ingress controller, the default ingress resource we create will not _"work"_  out of the box when the API endpoint is hit.

Comment (kmendiratta-crm) [reply modules/ROOT/pages/custom-ingress-configuration.adoc]: We can also perhaps consider providing a link to Kubernetes documentation for default ingressResource that we implement/create in our software by default.

File: modules/ROOT/pages/custom-ingress-configuration.adoc [hunk 6]
```diff
@@ -0,0 +1,52 @@
+= Defining a Custom Ingress Configuration
+
+[NOTE]
+====
+The tasks outline in this topic must be performed by an IT administrator
+====
+
+Runtime Fabric on Self-managed Kubernetes enables you to edit a template file that specifies a custom Ingress configuration. Using this file you can define annotations, rewrite rules, the ingressClassName, and more.
```
Comment (hegderohit89) [inline modules/ROOT/pages/custom-ingress-configuration.adoc]: Lets not add `add more`
Using this file you can define annotations, rewrite rules, and the ingressClassName.

As we don't support anything more at this phase, just don't want to give different expectations.

File: modules/ROOT/pages/custom-ingress-configuration.adoc [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ingress-config
+  namespace: rtf
+data:
+  ingress.annotations: |
+    nginx.ingress.kubernetes.io/rewrite-target: /
+    nginx.ingress.kubernetes.io/ssl-redirect: true
```
Comment (hegderohit89) [inline modules/ROOT/pages/custom-ingress-configuration.adoc]: `nginx.ingress.kubernetes.io/ssl-redirect: true` I am open to remove this line specifically. As we got some feed back from field where they said some use case requires the  `ssl-redirect: false` Or rather leave it this to the default implementation of ingress controller being used.

File: modules/ROOT/pages/custom-ingress-configuration.adoc [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+  ingress.spec: |
+    ingressClassName: <name>
+    rules.http.paths.path: /${APP_PATH}
+----
+
+. Edit the text file as required.
++
+All changes should be made under the `data` section of this file. Do not modify any other sections of the template. The most common configuration options are.
++
+* `ingress.annotations`: Annotations defined in this section are added to the `annotations` section of every Ingress resource created of a Mule application. Each annotation must be defined on a new line. Annotations defined here are default annotations that you can override.
```
Comment (hegderohit89) [inline modules/ROOT/pages/custom-ingress-configuration.adoc]: `Annotations defined here are default annotations that you can override.` --> Message what we need to convey is:
"These Annotations defined in the template are default kubernetes annotations, which can be fully overridden by user, if they wish to. They are not obligated to carry these default values".

Also can we make these as bullet points? 
Annotations defined in this section are added to the `annotations` section of every Ingress resource created of a Mule application. 
- Each annotation must be defined on a new line. 
- Annotations defined in the template are default kubernetes annotations, which can be fully overridden by user, if they wish to. They are not obligated to carry these default values.

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 4 above)
Comment (hegderohit89) [reply modules/ROOT/pages/custom-ingress-configuration.adoc:36]: Yeah with that we can clearly distinguish between Required and Optional vars

File: modules/ROOT/pages/custom-ingress-configuration.adoc [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+* `ingress.spec`: Used to configure the resource `spec` paramaters
+** `ingressClassName` (optional): Specifies the class name used in the Ingress resource created for a Mule application. Note, this is supported only on Kubernetes version 1.18 and higher.
```
Comment (hegderohit89) [inline modules/ROOT/pages/custom-ingress-configuration.adoc]: Also we should mention that its user responsibility to make sure that they are using the version of ingress controller which has support for this ingressClassName K8 feature.
Ex: I have seen few HAProxy community editions which still relies on `kubernetes.io/ingress.class: <name>` annotation even on k8 1.18 and higher clusters. 

So from our template side we support these users adding annotation and still getting it to work. 
cc: @kmendiratta-crm

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 1 above)
Comment (sleslie23) [reply modules/ROOT/pages/custom-ingress-configuration.adoc]: Changed.

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 4 above)
Comment (sleslie23) [reply modules/ROOT/pages/custom-ingress-configuration.adoc:37]: The problem is how do you represent the hierarchical relationship between elements? In the example you provided they are all at the same level.

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 2 above)
Comment (sleslie23) [reply modules/ROOT/pages/custom-ingress-configuration.adoc]: Changed.

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 3 above)
Comment (sleslie23) [reply modules/ROOT/pages/custom-ingress-configuration.adoc]: Fixed.

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 6 above)
Comment (sarjeet2013) [reply modules/ROOT/pages/custom-ingress-configuration.adoc]: Agree w/ Rohit. We can currently scope it to be limited custom changes that a customer can do to ingress configuration.

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 5 above)
Comment (sarjeet2013) [reply modules/ROOT/pages/custom-ingress-configuration.adoc]: Agree w/ Karan. Instead, we can clarify the following:
```If you do not apply a custom template file, Runtime Fabric uses a default configuration that may not work with the chosen ingress controller```

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 7 above)
Comment (sarjeet2013) [reply modules/ROOT/pages/custom-ingress-configuration.adoc]: Agree

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 4 above)
Comment (sarjeet2013) [reply modules/ROOT/pages/custom-ingress-configuration.adoc:36]: @sleslie23 They'll be at same level, so these can be defined in tabulated form.

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 8 above)
Comment (sarjeet2013) [reply modules/ROOT/pages/custom-ingress-configuration.adoc]: This looks OK to me. @hegderohit89 Do you have strong preference with your wording?

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 9 above)
Comment (sarjeet2013) [reply modules/ROOT/pages/custom-ingress-configuration.adoc]: We can clarify it better. how about this::
`ingressClassName` (optional): Specifies the class name used in the Ingress resource created for a Mule application. 
Note: ingressClassName within ingress spec is supported on Kubernetes version 1.18 and higher. Few ingress controllers may still use the annotations to define the ingressClassName.
Note: Ingress class name should be specified in ingress.annotations for Kubernetes version 1.17 and below, such as kubernetes.io/ingress.class: `<ingress-class-name>`

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 8 above)
Comment (hegderohit89) [reply modules/ROOT/pages/custom-ingress-configuration.adoc]: Not really, I am ok with the original text block too.  All i was saying is that we need to convey They are not obligated to carry these default values.

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 9 above)
Comment (hegderohit89) [reply modules/ROOT/pages/custom-ingress-configuration.adoc]: Yes, this looks perfect

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 6 above)
Comment (sleslie23) [reply modules/ROOT/pages/custom-ingress-configuration.adoc]: Changed.

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 5 above)
Comment (sleslie23) [reply modules/ROOT/pages/custom-ingress-configuration.adoc:12]: @sarjeet2013 I think that's too vague. We should at least tell them what to do if it doesn't work, which I assume is to create a custom configuration?

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 7 above)
Comment (sleslie23) [reply modules/ROOT/pages/custom-ingress-configuration.adoc:27]: This is the file I got from the spec. Please provide a revised version that contains exactly what we want the user modify.

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 4 above)
Comment (sleslie23) [reply modules/ROOT/pages/custom-ingress-configuration.adoc:36]: Will update based on revised template file in Google doc.

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 5 above)
Comment (sleslie23) [reply modules/ROOT/pages/custom-ingress-configuration.adoc]: Decided on the following text with Rohit and Karan:
- RTF tries to create a default ingress resource definition.
- Depending on your ingress controller, this resource definition may not work.
- In this case, you must apply a custom ConfigMap using the template.

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 7 above)
Comment (sleslie23) [reply modules/ROOT/pages/custom-ingress-configuration.adoc]: File is updated in the Google doc.

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 4 above)
Comment (sleslie23) [reply modules/ROOT/pages/custom-ingress-configuration.adoc:36]: Reformatted as a table.

File: modules/ROOT/pages/custom-ingress-configuration.adoc (hunk 8 above)
Comment (sleslie23) [reply modules/ROOT/pages/custom-ingress-configuration.adoc]: Leaving the original text.
